script/ide/System/Windows/Controls/Button_p.lua

```diff
@@ -184,38 +184,23 @@ function Button:paintNormalButton(painter)
 end
 
 function Button:paintCheckButton(painter)
+	self:CountLineList();
 	local size = self:GetSize();
 	local x, y, w, h = self:x() + (self:width() - size)/2, self:y() + (self:height() - size)/2, size, size;
 	
 	painter:SetPen("#9c9c9c");
-	--painter:SetPen(self.backgroundPens.check.outline_border);
 	painter:DrawRectTexture(x, y, w, h, "");
-
-	
---	local pointList = {{0,0},{0,w},{h,0},{w,h}};
---	painter:SetPen("#ffffff");
---	for i = 1, #pointList do
---		local pointX =  pointList[i][1];
---		local pointY =  pointList[i][2];
---		painter:DrawPoint(x + pointX, y + pointY);
---	end
-	
 			
 	painter:SetPen("#e0e0e0");
-	--painter:SetPen(self.backgroundPens.check.background);
 	painter:DrawRectTexture(x + 2, y + 2, w - 4, h - 4, "");
 
-
 	if(self.down or self.menuOpen or self.checked) then
 		painter:SetPen("#000000");
 		painter:SetPen({width = 2, color = "#000000"});
-		local lineList = {
-			{math.ceil(x + w*1/4), math.ceil(y + h/2), 0}, 
-			{math.ceil(x + w/2), math.ceil(y + h*3/4), 0}, 
-			{math.ceil(x + w/2), math.ceil(y + h*3/4), 0}, 
-			{math.ceil(x + w*3/4), math.ceil(y + h/4), 0}
-		};
-		painter:DrawLineList(lineList);
+
+		local lines = self.polygon_styles.check.lines;
+
+		painter:DrawLineList(lines);
```

Comment:
two triangles, might be better instead of lines 
![image](https://user-images.githubusercontent.com/94537/32033204-cc03ef32-ba3d-11e7-9406-579d8bfb6e3e.png)